salt_stack/salt/lambda-dev/files/makedomainenv

```diff
@@ -1,125 +1,86 @@
 
    # Fix spdb folder name.
    mv spdb.git spdb
    cd spdb
-   pip install -r requirements.txt
+   pip install -t ~/virtualenvs/$1/local/lib/python3.6/dist-packages -r requirements.txt
 
    cd c_lib/c_version
    cp makefile_LINUX makefile
    make
 
-   # (SH) When virtualenv fails, numpy is not listed in freeze.  This check will help debug the problem when it occurs again.
-   pip freeze | grep numpy
-   if [ $? -eq 0 ]
+   echo zipping up lambda and sending to s3
+   if [ -e ~/lambdazips/lambda.${1}.zip ]
    then
-       echo virtualenv installed correctly.
-       # removing debug files since virtualenv successfully installed
-       rm ~/debug/usr-before-${STARTTIME}.txt
-       rm  ~/debug/env-before-${STARTTIME}.txt
-       echo zipping up lambda and sending to s3
-       if [ -e ~/lambdazips/lambda.${1}.zip ]
-       then
-          rm ~/lambdazips/lambda.${1}.zip
-       fi
-       # we need to create a zip file with external symlinks removed but leave relative simlinks in place. We do 
-       # this because there is limit disk space in the lambda environment.  So we zip up virtualenv, which 
-       # converts all symlinks to underlying files.  Then we unzip and convert ./local and ./lib64 back to relative 
-       # symlinks, like there were originally (saving space), then we zip again with --symlinks option added, this 
-       # keeps the symlinks in place.     
-       if [ -e  ~/lambdazips/multilambda.${1}.zip ]; then
-          rm  ~/lambdazips/multilambda.${1}.zip
-       fi
-       cd ~/virtualenvs/${1}
-       zip -r -q  ~/lambdazips/multilambda.${1}.zip .
-
-       if [ -d ~/lambdazips/$1 ]; then
-          rm -rf ~/lambdazips/$1
-       fi
-       mkdir ~/lambdazips/$1
-       cd ~/lambdazips/$1
-       unzip -q ~/lambdazips/multilambda.${1}.zip
-       rm -rf lib64
-       ln -s lib lib64
-       rm -rf local
-       mkdir local
-       cd local
-       ln -s ../bin bin
-       ln -s ../include include
-       ln -s ../lib lib
-       ln -s ../lib64 lib64
-       chmod -R 775 ./lib/python3.4/site-packages
-       # fix put in because aws python2.7 now can't import a path with a period in it. (like python3.4)
-       cp lib64/python3.4/site-packages/lambda/lambda_loader.py .
-       cd ~/lambdazips/$1
-       rm ~/lambdazips/multilambda.${1}.zip
-       zip --symlinks -r -q  ~/lambdazips/multilambda.${1}.zip .
-
-       python ~/virtualenvs/${1}/lib/python3.4/site-packages/lambdautils/deploy_lambdas.py /home/ec2-user/virtualenvs/$1 ~/lambdazips/multilambda.${1}.zip $2 --upload-only
-   else
-       echo Virtualenv not working correctly.  Not creating lambda zip.
-       env > ~/debug/env-after-failing-${STARTTIME}.txt
-       find /usr -print > ~/debug/usr-after-failing-${STARTTIME}.txt
+      rm ~/lambdazips/lambda.${1}.zip
    fi
+
+   if [ -e  ~/lambdazips/multilambda.${1}.zip ]; then
+      rm  ~/lambdazips/multilambda.${1}.zip
+   fi
+
+   cd ~/virtualenvs/${1}/local/lib/python3.6/dist-packages
```

Comment:
Since we're using native Python 3.6 now, everything is put into the root of the .zip file so no path modifications required. Python packages have their top level folder placed in the root of the .zip.